CMakeLists.txt

```diff
@@ -7,6 +7,20 @@ add_subdirectory(src)
 
 option(LIBCOMMON_TESTS "Build the tests" ON)
 if (LIBCOMMON_TESTS)
+    # Download and unpack googletest at configure time
+    configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+    execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+            WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download"
+            )
+    execute_process(COMMAND "${CMAKE_COMMAND}" --build .
+            WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download"
+            )
+
+    # Add googletest directly to the build
+    add_subdirectory("${CMAKE_BINARY_DIR}/googletest-src"
+            "${CMAKE_BINARY_DIR}/googletest-build"
+            )
+
```

Comment:
Hat das einen Grund dass das nicht exakt das gleiche ist wie in https://github.com/google/googletest/blob/master/googletest/README.md#incorporating-into-an-existing-cmake-project ?
Da wird z.B. `CMAKE_CURRENT_BINARY_DIR` statt `CMAKE_BINARY_DIR` verwendet, und es gibt Fehlermeldungen.
```cmake
if(result)
  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
endif()
```

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_BINARY_DIR.html

Ich sehe, das könnte bewirken, dass wenn diese Lib als submodule (mit `add_subdirectory()`) genutzt wird, nur 1 googletest installation passiert statt 2. War das die intention? Wenn ja, bitte mit Kommentar dokumentieren.

Reply:
Primäre Intention war das nicht, ich bin lediglich nach dem Artikel [hier](https://crascit.com/2015/07/25/cmake-gtest/) gegangen, welcher vom Readme aus verlinkt war. Da war deutlich mehr Beschreibung drum rum

Reply:
Ich habe lediglich die Teile entfernt, die lediglich für ältere CMake-Versionen bzw. Windows notwendig waren

Reply:
Ach so, auch gut! Ich denke aber, zumindest die Fehlermeldungen könnte man integrieren, am Ende geht mal was schief, dann ist es gut die zu haben...

Reply:
Sollten jetzt drin sein.